src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,95 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
-
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
-
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
-        />
-      </span>
-    </p>
-  </form>
-);
+import React, { ChangeEvent, useState } from 'react';
+import { Todo } from '../../types/Todo';
+import { TodoList } from '../TodoList';
+
+type Props = {
+  todos: Todo[];
+};
+
+enum FilterStatus {
+  ALL = 'all',
+  ACTIVE = 'active',
+  COMPLETED = 'completed',
+}
+
+export const TodoFilter: React.FC<Props> = ({ todos }) => {
+  const [query, setQuery] = useState('');
+  const [filter, setFilter] = useState<FilterStatus>(FilterStatus.ALL);
+
+  const lowQuery = query.toLowerCase();
+
+  const filteredTodos = todos.filter((todo) => todo.title
+    .toLowerCase().includes(lowQuery)).filter((todo) => {
+    if (filter === FilterStatus.ACTIVE) {
```

Comment:
use one filter and handle logic with FilterStatus and lowQuery inside the callback